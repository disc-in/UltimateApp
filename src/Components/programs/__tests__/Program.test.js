import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import { createProgram, createTraining } from '../../../Fixtures/TestFixtures';
import store from '../../../Store/testStore';

import ConnectedProgram, { Program } from '../Program';

describe('<Program />', () => {
  let program = createProgram();

  it('renders correctly', () => {
    const { toJSON } = render(
      <Provider store={store}>
        <ConnectedProgram program={program} />
      </Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when finished', () => {
    const firstTraining = createTraining({ id: 1 });
    const secondTraining = createTraining({ id: 2 });
    program = createProgram({ trainings: [firstTraining, secondTraining] });

    const { toJSON } = render(
      <Program
        program={program}
        completeTrainings={[
          { training: firstTraining, program },
          { training: secondTraining, program },
        ]}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
