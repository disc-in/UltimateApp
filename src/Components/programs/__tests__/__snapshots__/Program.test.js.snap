// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Program /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "elevation": 3,
      "marginTop": 10,
      "padding": 10,
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#fff",
        "minHeight": 40,
        "paddingBottom": 10,
        "paddingHorizontal": 20,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#232b2b",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Beginner
      </Text>
      <Text
        style={
          Object {
            "color": "#232b2b",
            "fontSize": 14,
            "textAlign": "right",
          }
        }
      >
        0/1
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#f1f1f1",
          "borderRadius": 2,
          "height": 3,
          "marginTop": 5,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "backgroundColor": "#2a9e91",
              "borderRadius": 5,
              "width": "50%",
            },
            Object {
              "width": "0%",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`<Program /> renders correctly when finished 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "elevation": 3,
      "marginTop": 10,
      "padding": 10,
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#fff",
        "minHeight": 40,
        "paddingBottom": 10,
        "paddingHorizontal": 20,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#232b2b",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Beginner
      </Text>
      <Text
        style={
          Object {
            "color": "#232b2b",
            "fontSize": 14,
            "textAlign": "right",
          }
        }
      >
        2/2
         👍🎉
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#f1f1f1",
          "borderRadius": 2,
          "height": 3,
          "marginTop": 5,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "backgroundColor": "#2a9e91",
              "borderRadius": 5,
              "width": "50%",
            },
            Object {
              "width": "100%",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
